jaeger/charts/linkerd-jaeger/values.yaml

```diff
@@ -30,8 +30,39 @@ collector:
   # for more information
   tolerations: *default_tolerations
 
-  # -- address of the jaeger backend to send traces to
-  jaegerAddr:
+  # -- OpenTelemetry Collector config, See the
+  # [Configuration docs](https://opentelemetry.io/docs/collector/configuration/)
+  # for more information
+  # @default -- see `value.yaml` for actual configuration
+  config: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
+      opencensus:
+      zipkin:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_http:
+          thrift_compact:
+          thrift_binary:
+    processors:
+      batch:
+    extensions:
+      health_check:
+    exporters:
+      jaeger:
+        endpoint: jaeger.linkerd-jaeger:14250
```

Comment:
It makes sense to leave this part in the template to avoid having to warn users they need to keep in sync the `namespace` entry at the beginning of this file with the namespace used in this endpoint. Whether how much to leave here vs the template depends on how often would these entries be modified, over which I have no idea...

Reply:
Yep, I understand this issue, but Helm does not provide tools to use variables inside a `values.yml` file.

Maybe we can do the following trick.

First, add `POD_NAMESPACE` env variable into collector:
```
env:
  - name: POD_NAMESPACE
    valueFrom:
      fieldRef:
        apiVersion: v1
        fieldPath: metadata.namespace
```

Second, use the variable in config:
```
exporters:
  jaeger:
    endpoint: jaeger-collector.${POD_NAMESPACE}:14250
    insecure: true
```

We can do this because all components are installed in one namespace. If it is not then users should take care of the config anyway and sync endpoints.

@Pothulapati @alpeb What do you think?

Reply:
That sounds good to me. And then we can keep the whole config in values.yaml :+1:

Reply:
@Pothulapati @alpeb I've made the necessary changes. Hope we can merge PR 😃